app/models/force/object_base.rb

```diff
@@ -0,0 +1,42 @@
+module Force
+  # Base representation of a Salesforce object. Provide mapping between
+  # SOQL API and LAP domain field names for preferences.
+  class ObjectBase
+    def initialize(fields, format)
+      raise ArgumentError, 'Field values and format are required.' unless fields && format
+
+      @fields = {
+        soql: {},
```

Comment:
As I've mentioned to everyone in person, "soql" here is kind of a misnomer. Although we use this set of field names in SOQL queries, the field names themselves have nothing to do with SOQL at all. They are the field names on the Salesforce objects. A task for us as a team going forward is to come up with better terminology to refer to "the set of field names on Salesforce objects"!

Reply:
are there separate names for the restforce API and for SOQL?

If not, what if we just call it salesforce? or salesforce_field

so then we could have

domain
salesforce
custom_api?

Reply:
the salesforce standard REST API that we use through the restforce gem uses the same field names as SOQL does, so yeah there's no difference there in terms of field naming. you know, just "salesforce" as a name isn't bad. it's still a little bit confusing because the custom API is also a way of interacting with salesforce. but it's closer to being accurate than "soql" is. i'd be down to change "soql" to "salesforce" here as a step in the right direction, and i like "custom_api" as a slight improvement over just "custom", and then as we continue to move forward with this we can continue to improve our naming system.